examples/elixir/udp_hole_punching/rendezvous_worker.exs

```diff
@@ -0,0 +1,75 @@
+defmodule RendezvousWorker do
+  @moduledoc """
+  An Ockam Worker that acts as a rendezvous server for UDP hole punching.
+  """
+
+  use Ockam.Worker
+
+  require Logger
+
+  alias Ockam.{Message, Router}
+
+  @impl true
+  def handle_message(%{payload: "ping"} = message, state) do
+    Router.route(Message.reply(message, state.address, "pong"))
+
+    {:ok, state}
+  end
+
+  def handle_message(%{payload: "my address"} = message, state) do
+    [external_address, source] = message.return_route
+
+    Logger.debug("Replying with address: #{external_address.value}")
+
+    Router.route(Message.reply(message, state.address, "address: #{external_address.value}"))
+
+    state = put_in(state, [:attributes, :addresses, source], external_address)
+    {:ok, state}
+  end
+
+  def handle_message(%{payload: "connect"} = message, state) do
+    source = message.return_route |> Enum.reverse() |> hd()
+    target = message.onward_route |> Enum.reverse() |> hd()
+
+    Logger.debug("Received connect message from #{inspect(source)} to #{inspect(target)}")
+
+    state =
+      state.attributes.addresses
+      |> Map.get(target)
+      |> case do
+        nil ->
+          Logger.debug("Target #{target} not found")
+          pending = [{source, target} | state.attributes.pending_connections]
+          put_in(state, [:attributes, :pending_connections], pending)
```

Comment:
I understand the intention is to let unmatched connection request on the `:pending_connections`  list,  but that is never used latter on,   there is some missing  there.